src/webhook/namespace_validation.py

```diff
@@ -40,25 +40,45 @@ def handle_request():
 def get_response(req, debug=False):
   try:
     body_dict = req.json['request']
-    requester_group_memberships = []
-    if 'userInfo' in body_dict:
-      if 'groups' in body_dict['userInfo']:
-        requester_group_memberships = body_dict['userInfo']['groups']
-    else:
-      return responses.response_invalid()
-    if "dedicated-admins" in requester_group_memberships:
-      requested_ns = body_dict['namespace']
-      privileged_namespace_re = '(^kube-.*|^openshift.*|^ops-health-monitoring$|^management-infra$|^default$|^logging$|^sre-app-check$|^redhat-.*)'
-      # match will return a match object if the namespace matches the regex, or None if the namespace doesn't match the regex
-      if re.match(privileged_namespace_re, requested_ns) is not None:
-        DENIED_NAMESPACE.inc()
-        return responses.response_deny(req=body_dict, msg="You cannot update the privileged namespace {}.".format(requested_ns))
-      else:
+    userinfo = body_dict['userInfo']
+    requester_group_memberships = userinfo['groups']
+    privileged_namespace_re = '(^kube.*|^openshift.*|^default$|^redhat.*)'
+    requested_ns = body_dict['namespace']
+    cluster_admin_users = ["kube:admin", "system:admin"]
+
+    # check to see if requester is a serviceAccount in a privileged NS
+    # if so, SA can edit any namespace
+    # re.match will return a match object if the namespace matches the regex,
+    # or None if the string doesn't match the regex
+    privileged_serviceaccount_re = '^system:serviceaccounts:(kube.*|openshift.*|default|redhat.*)'
+    for group in requester_group_memberships:
+      if re.match(privileged_serviceaccount_re, group) is not None:
         return responses.response_allow(req=body_dict)
```

Comment:
@jewzaam This is well past merge, but it looks like this is matching the service account name to user group names? Notably, this section of code does not have tests, which could help with the previous question

Reply:
I talked to Lisa about this a little earlier today, since this is my code. She's right that there's no unit tests for this bit. I'll create a PR with unit tests for it.